test/test_lazy_clone.py

```diff
@@ -239,13 +385,75 @@ def test_clone_after_lazy_clone(self, device):
         self.assertTrue(torch._C._is_cow_tensor(a))
         self.assertTrue(torch._C._is_cow_tensor(b))
         self.assertFalse(torch._C._is_cow_tensor(c))
-        self.assertEqual(torch._C._data_address_resolve_unified(a), orig_data_ptr)
-        self.assertEqual(torch._C._data_address_resolve_unified(b), orig_data_ptr)
+        self.assertEqual(data_address_unified(a), orig_data_ptr)
+        self.assertEqual(data_address_unified(b), orig_data_ptr)
 
         self.assertEqual(a, b)
         self.assertEqual(a, c)
         self.assertEqual(b, c)
 
+    @skipCUDAIf(True, "Does not work for CUDA")
+    @skipXLA
+    def test_to_compile(self, device):
+        pin_memory = torch.device(device).type == "mps"
+
+        @torch.compile
+        def fn(x):
+            x = x + 1
+            x = x + 2
+            x = x.to(device=device)
+            x = x + 3
+            x = x + 4
+            x = x.cpu()
+            x = x + 5
+            x = x + 6
+            x = x.to(device=device)
+            x = x + 7
+            x = x + 8
+            x = x.cpu()
+            x = x + 9
+            x = x + 10
+            return x
+
+        fn(torch.randn([2, 2, 10], pin_memory=pin_memory))
+
+    # See Note [CPU pinned to MPS failures]
+    # TODO: Once this issue is fixed, remove this test
+    def test_isclose_issue(self, device):
+        a_device = torch.device(device).type
+        b_device = "mps" if a_device == "cpu" else "cpu"
+
+        a_args = [
+            torch.randint(
+                -9,
+                10,
+                (5, 10, 5),
+                dtype=torch.int32,
+                device=a_device,
+                pin_memory=a_device == "cpu",
+            ),
+            torch.randint(
+                -9,
+                10,
+                (5, 10, 5),
+                dtype=torch.int32,
+                device=a_device,
+                pin_memory=a_device == "cpu",
+            ),
+        ]
+
+        b_args = [arg.to(b_device) for arg in a_args]
+
+        # This op call mutates the first arg's data
+        if a_device == "mps":
+            torch.isclose(*a_args)
+        else:
+            torch.isclose(*b_args)
+
+        self.assertEqual(a_args[1], b_args[1])
+        # THIS one fails
+        self.assertEqual(a_args[0], b_args[0])
```

Comment:
I'm having a really hard time trying to fix this failure. If anyone has any insights, please let me know. As far as I know, this is the last thing I need to fix before this can go public